core/src/main/scala/com/softwaremill/react/kafka_api_opt3/Example.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.softwaremill.react.kafka_api_opt3
+
+import scala.concurrent.Future
+
+import akka.Done
+import com.softwaremill.react.kafka_api_producer.Producer
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+
+trait Example {
+  implicit val ec = scala.concurrent.ExecutionContext.Implicits.global
+
+  def consumerProvider: () => KafkaConsumer[Array[Byte], String] = ???
+
+  def producerProvider: () => KafkaProducer[Array[Byte], String] = ???
+
+  class DB {
+    def save(offset: Long, data: String): Future[Done] = ???
+  }
+}
+
+// Consume messages and store a representation, including offset, in DB
+object Example3 extends Example {
+  val db = new DB
+
+  Consumer.source(consumerProvider)
```

Comment:
I assume that `Source` manages kafka consumer lifecycle. When I call `stop` what will happens to kafka consumer? We can not stop it because we have to be able to handle their commit and we have no idea when all these messages processed.